spark3/src/main/java/org/apache/iceberg/actions/Spark3CreateAction.java

```diff
@@ -0,0 +1,290 @@
+    long numMigratedFiles = 0;
+    try {
+      SparkTableUtil.importSparkTable(spark, Spark3Util.toTableIdentifier(sourceTableName), icebergTable,
+          stagingLocation);
+
+      Snapshot snapshot = icebergTable.currentSnapshot();
+      numMigratedFiles = Long.valueOf(snapshot.summary().get(SnapshotSummary.TOTAL_DATA_FILES_PROP));
+
+      stagedTable.commitStagedChanges();
+      LOG.info("Successfully loaded Iceberg metadata for {} files", numMigratedFiles);
+    } catch (Exception e) {
+      LOG.error("Error when attempting to commit migration changes, rolling back", e);
+      stagedTable.abortStagedChanges();
+      throw e;
+    }
+
+    if (sessionCatalogReplacement) {
+      Identifier replacementTable = Identifier.of(destTableName.namespace(), destTableName.name() + REPLACEMENT_NAME);
+      try {
+        stagingCatalog.dropTable(destTableName);
+        stagingCatalog.renameTable(replacementTable, destTableName);
+      } catch (NoSuchTableException e) {
+        LOG.error("Cannot migrate, replacement table is missing. Attempting to recreate source table", e);
+        try {
+          stagingCatalog.createTable(sourceTableName, sourceTable.schema(),
+              Spark3Util.toTransforms(sourcePartitionSpec), JavaConverters.mapAsJavaMap(sourceTable.properties()));
+        } catch (TableAlreadyExistsException tableAlreadyExistsException) {
+          Log.error("Cannot recreate source table. Source table has already been recreated", e);
+          throw new RuntimeException(e);
+        } catch (NoSuchNamespaceException noSuchNamespaceException) {
+          Log.error("Cannot recreate source table. Source namespace has been removed, cannot recreate", e);
+          throw new RuntimeException(e);
+        }
+      } catch (TableAlreadyExistsException e) {
+        Log.error("Cannot migrate, Source table was recreated before replacement could be moved. " +
+            "Attempting to remove replacement table.", e);
+        stagingCatalog.dropTable(replacementTable);
+        stagedTable.abortStagedChanges();
+      }
+    }
+
+    return numMigratedFiles;
+  }
+
+  @Override
+  public CreateAction withAdditionalProperties(Map<String, String> properties) {
+    this.additionalProperties.putAll(properties);
+    return this;
+  }
+
+  @Override
+  public CreateAction withAdditionalProperty(String key, String value) {
+    this.additionalProperties.put(key, value);
+    return this;
+  }
+
+  @Override
+  public CreateAction as(String newName) {
+    Spark3Util.CatalogAndIdentifier newDest = Spark3Util.catalogAndIdentifier(spark, newName);
+    return new Spark3CreateAction(spark, sourceCatalog, sourceTableName, newDest.catalog(), newDest.identifier())
+        .withAdditionalProperties(this.additionalProperties);
+  }
+
+  private static void validateSourceTable(CatalogTable sourceTable, Set<String> supportedSourceTableProviders) {
+    String sourceTableProvider = sourceTable.provider().get().toLowerCase(Locale.ROOT);
```

Comment:
This excludes Hive tables that don't use Spark's provider, including those created with `STORED AS parquet` instead of `USING parquet`. I don't think that is necessary. Isn't the only requirement that all of the partitions are a supported format?

Reply:
Yeah let me double check this.

Reply:
So this is ok, Managed tables return a provider of hive.

![image](https://user-images.githubusercontent.com/413025/99304613-33e67700-2818-11eb-8f25-03cfa32e5fef.png)

Reply:
Also added managed hive table tests